### PR TITLE
Add kubernetes-e2e-gce-release-1.2 e2e suite.

### PR DESCRIPTION
**Pull request by spxtr**
I want to make sure I got this project set up properly before doing the rest. Resulting XML looks good.

ref #22672

### CODE REVIEW

File: hack/jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e.yaml [hunk 1]
```diff
@@ -226,6 +226,24 @@
         - 'kubernetes-e2e-{suffix}'
 
 - project:
+    name: kubernetes-e2e-gce-1-2
```
Comment (ixdy) [inline hack/jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e.yaml]: is there any meaning to the order of projects in this file?

File: hack/jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e.yaml [hunk 2]
```diff
@@ -226,6 +226,24 @@
...
+    trigger-job: 'kubernetes-build-1.2'
+    test-owner: 'Build Cop'
+    provider-env: |
+        {gce-provider-env}
+        export JENKINS_PUBLISHED_VERSION="ci/latest-1.2"
```
Comment (ixdy) [inline hack/jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e.yaml]: putting this in 'provider-env' seems somewhat arbitrary, but I guess there's no better place?

seems like we may want to have this version as another option for projects? can default to 'ci/latest', but override to 'ci/latest-1.2' here, 'ci/latest-1.1' for 1.1 builds, etc?

File: hack/jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e.yaml [hunk 3]
```diff
@@ -226,6 +226,24 @@
         - 'kubernetes-e2e-{suffix}'
 
 - project:
+    name: kubernetes-e2e-gce-1-2
+    trigger-job: 'kubernetes-build-1.2'
```
Comment (ixdy) [inline hack/jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e.yaml]: should we also set `branch`, even though it's not used by e2e jobs as best I can tell?

File: hack/jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e.yaml (hunk 2 above)
Comment (spxtr) [inline hack/jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e.yaml]: Yes, I want to wait until we've got the 1.2 jobs in then do a big YAML cleanup. We can start doing things like putting `export JENKINS_PUBLISHED_VERSION="{published-version}"` in the build script along with a global default to `ci/latest` and stuff. It'll be nice.

File: hack/jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e.yaml (hunk 3 above)
Comment (spxtr) [inline hack/jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e.yaml]: No, I should remove the branch variable from the others. It used to be used to call the runner for that branch, back when they were all `e2e.sh`.

File: hack/jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e.yaml (hunk 1 above)
Comment (spxtr) [inline hack/jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e.yaml]: Not really. I put this in between the master and 1.1 jobs, but there are master branch feature tests elsewhere and it's a bit of a mess.

File: hack/jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e.yaml (hunk 2 above)
Comment (ixdy) [inline hack/jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e.yaml]: :+1:

File: hack/jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e.yaml [hunk 4]
```diff
@@ -226,6 +226,24 @@
...
+    test-owner: 'Build Cop'
+    provider-env: |
+        {gce-provider-env}
+        export JENKINS_PUBLISHED_VERSION="ci/latest-1.2"
+    suffix:
+        - 'gce-release-1.2':
+            description: 'Runs all non-slow, non-serial, non-flaky, tests on GCE in parallel.'
```
Comment (ixdy) [inline hack/jenkins/job-configs/kubernetes-jenkins/kubernetes-e2e.yaml]: Description should maybe mention release-1.2?

**Comment by ixdy** [comment]
Basically LGTM. Not tagging since I'm guessing we want to manually merge.

**Comment by spxtr** [comment]
Updated the description.

Going to manually merge to watch Jenkins.

**Pull request merged**